File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.cpp
class Solution {
public:
  int characterReplacement(string s, int k) {
    int res = 0;
    for (int i = 0; i < s.size(); i++) {
      unordered_map<char, int> count(26);
      int maxf = 0;
      for (int j = i; j < s.size(); j++) {
        maxf = max(maxf, count[s[j]]++);
        if ((j - i )-maxf <= k) {
          res = max(res, (j - i + 1));
        }
      }
    }
    return res;
  }

};

File: Data Structures & Algorithms/anagram-groups/submission-1.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> res = new HashMap<>();

        for (String str : strs) {
            char[] count = new char[26];
            for (char c : str.toCharArray()) {
                count[c - 'a']++;
            }
            String key = new String(count);
            if (!res.containsKey(key)) {
                res.put(key, new ArrayList<>());
            }
            res.get(key).add(str);
        }

        return new ArrayList<>(res.values());
    }
}